File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dic = {}
    
    for name, kind in clothes:
        if kind in dic.keys():
            dic[kind] += [name]
        else:
            dic[kind] = [name]
    
    for kind, name in dic.items():
        answer *= len(name) + 1
        
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, List<String>> clothMap = new HashMap<>();

        for (int i = 0; i < clothes.length; i++) {
            String name = clothes[i][0];
            String type = clothes[i][1];
            
            if (!clothMap.containsKey(type)) {
                clothMap.put(type, new ArrayList<>());
            }
            clothMap.get(type).add(name);
        }
        
        
        int answer = 1; // default
        for (List<String> c : clothMap.values()) {
            answer *= (c.size() + 1);
        }
        
        return answer - 1;
    }
}